### ISSUE TITLE
Export MediaTailorAdManager and add ability to set tracking URL

### ISSUE THREAD

**Issue by cristian-atehortua** [shaka-project/shaka-player#6234]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes, there is nothing related to this

**Is your feature request related to a problem? Please describe.**
Our app uses MediaTailor Server side ads but didn't call the endpoint to get the manifest and the tracking URL. Instead, we receive a manifest URL from our backend that, when `GET`ting it, it replies with a redirection to the MediaTailor manifest. It makes the default `AdManager` unusable for us because when used with MediaTailor it expects that the provided URL replies with a JSON file and not directly with the manifest.

**Describe the solution you'd like**
Export the `MediaTailorAdManager` on the build so it can be used in the `Player.setAdManagerFactory`, and add a method to manually set the tracking URL and update the ad data.

**Describe alternatives you've considered**
Create my own AdManager copying logic from MediaTailorAdManager.

**Additional context**

**Comment by avelad** [comment]
@cristian-atehortua Can you review if https://github.com/shaka-project/shaka-player/pull/6365 helps?

We do not want to export any ad manager by default, but it seems reasonable to state the case by only adding the tracking url.

The PR code would be used as follows:

```
const adManager = this.player_.getAdManager();
const container = this.controls_.getServerSideAdContainer();
const netEngine = this.player_.getNetworkingEngine();
adManager.initMediaTailor(container, netEngine, this.video_);
adManager.addMediaTailorTrackingUrl(YOUR_TRACKING_URL);
this.player_.load(YOUR_STREAM_URL);
```

**Comment by cristian-atehortua** [comment]
Hi @avelad. Thanks for your reply and the PR, it perfectly solves my request and I think is probably better than exporting a new ad manager (I realized that time after creating the issue). It's more consistent with the current design.

I also realized that, from my point of view, it would be better if the `IAdManager` interface defines only the functions that Shaka uses or may use. For me, it doesn't make sense that a custom ad manager that has nothing to do with MediaTailor, for example, has to implement `initMediaTailor` or `requestMediaTailorStream`. And, on the other side, it doesn't make sense to have to change the interface when adding a function that Shaka doesn't use directly.

Don't you think the same?
Of course, it's a matter of discussion and design, but I think the design is more robust in that way.
